Инфа о пул реквесте

Dmitriy-Evs opened the pull request.

Dmitriy-Evs committed:
second commit

mkouznetsov-git commented on src/test/java/ScooterTest/FaqTest.java:
```diff
@@ -0,0 +1,108 @@
+
+    @Before
+    public void setIp(){
+        System.setProperty("webdriver.gecko.driver", "C:\\Study\\geckodriver.exe");
```
Не у всех драйвер расположен в таком месте. И не у всех даже винда стоит на компе. )
Лучше просто использовать дефолтный драйвер, установленный в системе:
`driver = new FirefoxDriver();`
или
`driver = new ChromeDriver();`

mkouznetsov-git commented on src/test/java/ScooterTest/FaqTest.java:
```diff
@@ -0,0 +1,108 @@
+package ScooterTest;
+
+/*import io.github.bonigarcia.wdm.WebDriverManager;*/
+import org.hamcrest.MatcherAssert;
```
Импорт не используется

mkouznetsov-git commented on src/test/java/ScooterTest/FaqTest.java:
```diff
@@ -0,0 +1,108 @@
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.openqa.selenium.By;
```
Импорт не используется

mkouznetsov-git commented on src/test/java/ScooterTest/FaqTest.java:
```diff
@@ -0,0 +1,108 @@
+import org.junit.Before;
+import org.junit.Test;
+import org.openqa.selenium.By;
+import org.openqa.selenium.JavascriptExecutor;
```
Импорт не используется

mkouznetsov-git commented on src/test/java/ScooterTest/FaqTest.java:
```diff
@@ -0,0 +1,108 @@
+import org.openqa.selenium.By;
+import org.openqa.selenium.JavascriptExecutor;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
```
Импорт не используется

mkouznetsov-git commented on src/test/java/ScooterTest/FaqTest.java:
```diff
@@ -0,0 +1,108 @@
+import org.openqa.selenium.JavascriptExecutor;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.chrome.ChromeDriver;
```
Импорт не используется

mkouznetsov-git commented on src/test/java/ScooterTest/FaqTest.java:
```diff
@@ -0,0 +1,108 @@
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.chrome.ChromeDriver;
+import org.openqa.selenium.firefox.FirefoxDriver;
+import org.openqa.selenium.support.ui.ExpectedConditions;
```
Импорт не используется

mkouznetsov-git commented on src/test/java/ScooterTest/FaqTest.java:
```diff
@@ -0,0 +1,108 @@
+import java.time.Duration;
+import java.util.concurrent.TimeUnit;
+
+import static java.awt.SystemColor.window;
```
Импорт не используется

mkouznetsov-git commented on src/test/java/ScooterTest/FaqTest.java:
```diff
@@ -0,0 +1,108 @@
+import java.util.concurrent.TimeUnit;
+
+import static java.awt.SystemColor.window;
+import static org.hamcrest.CoreMatchers.is;
```
Импорт не используется

mkouznetsov-git commented on src/test/java/ScooterTest/FaqTest.java:
```diff
@@ -0,0 +1,108 @@
+    }
+
+    @Test
+    public void checkFaq(){
```
По сути, у тебя тут 8 одинаковых тестовых методов. Параметризуй вопросы-ответы и сделай один тестовый метод. Следует всегда стараться придерживаться принципа DRY.

mkouznetsov-git commented on src/test/java/ScooterTest/scooterOrderTest.java:
```diff
@@ -0,0 +1,84 @@
+    }
+
+    @Test
+    public void orderScooter(){
```
А здесь надо параметризовать кнопку заказа (верхняя/нижняя) и также оставить один тестовый метод.

Dmitriy-Evs committed:
1. Для вопросов ответов используется один метод, на вход которого я передаю необходимый локатор
2. Написал функцию для вызова кнопки "Заказать", чтобы в классе теста был только один метод, в который передается нужная кнопка
3. Удалил все неиспользуемые импорты
4. Поправил вызов браузера FireFox. Изначальный способ с файлом на компе был использован из-за того, что иначе была ошибка вызова браузера.

mkouznetsov-git commented on src/test/java/ScooterTest/FaqTest.java:
```diff
@@ -0,0 +1,97 @@
+            mainPage.clickCookieButton();
+
+        new WebDriverWait(driver, Duration.ofSeconds(1));
+
```
Не совсем то. Сделай один короткий тест на проверку одного вопроса-ответа, а в параметрах задай список этих вопросов-ответов с ожидаемыми значениями текста, с локаторами элементов или порядковыми номерами, если будешь делать отдельный метод определения локаторов по индексу.

mkouznetsov-git commented on src/test/java/ScooterTest/scooterOrderTest.java:
```diff
@@ -0,0 +1,82 @@
+import java.util.concurrent.TimeUnit;
+import static org.junit.Assert.assertTrue;
+
+public class scooterOrderTest {
```
Классы надо называть с большой буквы.

mkouznetsov-git commented on src/test/java/ScooterTest/scooterOrderTest.java:
```diff
@@ -0,0 +1,82 @@
+        OrderPage orderPage = new OrderPage(driver);
+
+        mainPage.clickCookieButton();
+        mainPage.clickOrderButton("Верхняя кнопка");
```
Используй параметризацию -- туда можно засунуть кнопку заказа и все наборы тестовых данных. И сделай один проверочный метод. Он у тебя сейчас в двух экземплярах, это лишнее.

Dmitriy-Evs committed:
1. Изменил название класса на корректное.
2. Параметризировал все, что смог. Получилось полностью это сделать со сценарием заказа. Но ответы на вопросы мне не поддались, к тому сегодня не у кого спросить. У меня получилось параметризировать вопросы, но с получением текста ответа возникили сложности. Так что, запушу рабочую версию.

mkouznetsov-git commented on src/test/java/ScooterTest/FaqTest.java:
```diff
@@ -0,0 +1,97 @@
+        driver.manage().timeouts().implicitlyWait(5, TimeUnit.SECONDS);
+        driver.get("https://qa-scooter.praktikum-services.ru/");
+    }
+
```
Понимаешь, такой подход, когда у тебя в одном тестовом методе куча проверок, ещё плох тем, что если у тебя свалится в начале какая-то проверка, то на этом всё закончится, тест выдаст ошибку и остальные сценарии не будут проверены.

mkouznetsov-git commented on src/test/java/ScooterTest/FaqTest.java:
```diff
@@ -0,0 +1,97 @@
+import java.util.concurrent.TimeUnit;
+import static org.junit.Assert.assertEquals;
+
+public class FaqTest {
```
По твоему вопросу о параметризации -- пример, как это можно сделать:

    public class FaqTest extends BaseTest{
        private final int elementIndex;
        private final String expectedText;

        public FaqTest(int index, String answerText) {
            this.elementIndex = index;
            this.expectedText = answerText;
        }

        @Parameterized.Parameters()
        public static Object[][] getInput() {
            return new Object[][]{
                    {0, "Сутки — 400 рублей. Оплата курьеру — наличными или картой."},
                    {1, "Пока что у нас так: один заказ — один самокат. Если хотите покататься с друзьями, можете просто сделать несколько заказов — один за другим."},
                    {2, "Допустим, вы оформляете заказ на 8 мая. Мы привозим самокат 8 мая в течение дня. Отсчёт времени аренды начинается с момента, когда вы оплатите заказ курьеру. Если мы привезли самокат 8 мая в 20:30, суточная аренда закончится 9 мая в 20:30."},
                    {3, "Только начиная с завтрашнего дня. Но скоро станем расторопнее."},
                    {4, "Пока что нет! Но если что-то срочное — всегда можно позвонить в поддержку по красивому номеру 1010."},
                    {5, "Самокат приезжает к вам с полной зарядкой. Этого хватает на восемь суток — даже если будете кататься без передышек и во сне. Зарядка не понадобится."},
                    {6, "Да, пока самокат не привезли. Штрафа не будет, объяснительной записки тоже не попросим. Все же свои."},
                    {7, "Да, обязательно. Всем самокатов! И Москве, и Московской области."},
            };
        }
    
        @Test
        public void checkFaq() {
    ... здесь проверочный метод, который по номеру элемента определяет текущий текст (это лучше вынести в отдельный метод) и сравнивает с ожидаемым из параметров ...
        }

Также обрати внимание, что все локаторы вопросов-ответов одинаковые и различаются только цифрой в конце -- это удобно тем, что можно заменяя одну цифру (по сути, порядковый номер), собирать нужный нам локатор:
".//div[@id = 'accordion__panel-0']"
".//div[@id = 'accordion__panel-1']"
...
".//div[@id = 'accordion__panel-7']"

Dmitriy-Evs committed:
1. Сделал параметризацию для теста вопросов-ответов, теперь там один метод.
2. Еще раз почистил лишние импорты.

Dmitriy-Evs committed:
1. Поправил импорты (случайно удалил лишние в предыдущем коммите)
2. Добавил ожидание.
